From 3e04d944626ba4caee8a4d74a683c6c967e935ca Mon Sep 17 00:00:00 2001
From: David Henningsson <david.henningsson@canonical.com>
Date: Tue, 13 Aug 2013 12:43:29 +0200
Subject: [PATCH 203/204] card: Add hook before profile changes

This is useful to modules that want to do things just before a
profile becomes inactive.
---
 src/pulsecore/card.c |    2 ++
 src/pulsecore/core.h |    1 +
 2 files changed, 3 insertions(+)

Index: pulseaudio/src/pulsecore/card.c
===================================================================
--- pulseaudio.orig/src/pulsecore/card.c
+++ pulseaudio/src/pulsecore/card.c
@@ -276,6 +276,8 @@ int pa_card_set_profile(pa_card *c, pa_c
         return 0;
     }
 
+    pa_hook_fire(&c->core->hooks[PA_CORE_HOOK_CARD_PROFILE_CHANGING], profile);
+
     if ((r = c->set_profile(c, profile)) < 0)
         return r;
 
Index: pulseaudio/src/pulsecore/core.h
===================================================================
--- pulseaudio.orig/src/pulsecore/core.h
+++ pulseaudio/src/pulsecore/core.h
@@ -122,6 +122,7 @@ typedef enum pa_core_hook {
     PA_CORE_HOOK_CARD_PUT,
     PA_CORE_HOOK_CARD_UNLINK,
     PA_CORE_HOOK_CARD_PROFILE_CHANGED,
+    PA_CORE_HOOK_CARD_PROFILE_CHANGING,
     PA_CORE_HOOK_CARD_PROFILE_ADDED,
     PA_CORE_HOOK_CARD_PROFILE_AVAILABLE_CHANGED,
     PA_CORE_HOOK_PORT_AVAILABLE_CHANGED,
